### ISSUE TITLE
Support for reorder-python-imports

### ISSUE THREAD

**Issue by alexdauenhauer** [charliermarsh/ruff#4670]
While I personally like `isort` better, my company uses [reorder-python-imports](https://pypi.org/project/reorder-python-imports/) so I am wondering if there is already support for this and I missed it in the docs, or if support could be added? I am loving this package and would love to have it auto-run the import sorting in the format I need.

**Comment by alexdauenhauer** [comment]
actually it turns out I can almost achieve the same functionality, but the isort arg that I need is `force-alphabetical-sort-within-sections` and ruff does not yet support this

**Comment by JonathanPlasse** [comment]
Can you share the configuration?
A profile `reporder-python-imports` could be added.

**Comment by alexdauenhauer** [comment]
well now I think what would actually be better than adding support for reorder-python-imports, would be to just add the `force-alphabetical-sort-within-sections` flag for available `isort` args

**Comment by alexdauenhauer** [comment]
@sanmai-NL unfortunately not. The sorting rules are slightly different. sort-within-sections prioritizes case over alphanumeric (e.g. `MY_CONSTANT` will be sorted on top of `alphabetical_method`)

**Comment by perezzini** [comment]
Any updates about this? Thanks!

**Comment by owenlamont** [comment]
I'd like this too, I'm trying to get Ruff adopted where I work but this is one area where a lot of code gets formatted differently to our current isort configuration.

**Comment by owenlamont** [comment]
Update: Actually I don't think this is an issue anymore. Once I noticed the _case-sensitive_ argument which I set to _false_ and I fixed and tweaked some other ruff configuration I was able to exactly reproduce my old isort configuration sorting behaviour that had force_alphabetical_sort_within_sections set to true.

**Comment by Pierre-Sassoulas** [comment]
@owenlamont do you mind sharing your configuration to mimic `force_alphabetical_sort_within_sections`, please ? I didn't manage to recreate it from the description.

**Comment by owenlamont** [comment]
Hi @Pierre-Sassoulas - this was my Ruff isort config taken from my pyproject.toml (note I excluded known-first-party and known-third-party modules which will be specific to your repo). This sorting was close to but not quite identical to isort. I think there were some minor differences with Ruff sorting on full relative import paths (which I personally liked) - I think the difference was something like that - I don't exactly remember but most of the time the import sort order came out identical to how we had isort configured.

```toml
[tool.ruff.isort]
case-sensitive = false
combine-as-imports = true
force-sort-within-sections = true
known-first-party = []
known-third-party = []
lines-after-imports = 2
order-by-type = false
section-order = [
    "future",
    "standard-library",
    "third-party",
    "first-party",
    "local-folder"
]
```

This closely matched the behaviour of the old isort config:

```toml
[tool.isort]
profile = "black"
multi_line_output = 3
force_sort_within_sections = true
force_alphabetical_sort_within_sections = true
sections = ["FUTURE", "STDLIB", "THIRDPARTY", "FIRSTPARTY", "LOCALFOLDER"]
known_first_party = []
known_thirdparty = []
skip_gitignore=true
lines_after_imports=2
combine_as_imports=true
```

**Comment by Pierre-Sassoulas** [comment]
It's working, thank you a lot @owenlamont !

> The sorting rules are slightly different. sort-within-sections prioritizes case over alphanumeric (e.g. MY_CONSTANT will be sorted on top of alphabetical_method)

The specific option that fix this is ``order-by-type = false``.

I guess the issue can be closed then.

**Comment by Jari27** [comment]
I have a similar problem that I cannot seem to solve with the current settings.  

I have the following 5 imports:
```python
from pyspark.sql import Column, DataFrame
from pyspark.sql import functions as F
from pyspark.sql import SparkSession
from pyspark.sql import types as T
```

These are sorted by isort with the following config:
```ini
[tool.isort]
line_length = 120
multi_line_output = 3
force_alphabetical_sort_within_sections = "True"
force_sort_within_sections = "False"
profile = "black"
```

I cannot seem to get the same behaviour with ruff, as it always performs the sorting like this:
```python
from pyspark.sql import Column, DataFrame, SparkSession
from pyspark.sql import functions as F
from pyspark.sql import types as T
```

with the following config:
```ini
[tool.ruff.lint.isort]
force-sort-within-sections = false
order-by-type = false
case-sensitive = false
```

Am I misconfiguring anything? It seems like the alphabetical sort is sorting AFTER grouping, instead of before. And I cannot seem to turn this off.

**Comment by Jari27** [comment]
@samdoran Thanks, that is a lot closer but still not fully compatible. isort (`isort==5.10.1`) instead puts `Column` and `DataFrame` on the same line.

**Comment by charliermarsh** [comment]
Yes we don't support that.

**Comment by charliermarsh** [comment]
It's intentional and documented here: https://docs.astral.sh/ruff/faq/#how-does-ruffs-import-sorting-compare-to-isort

**Comment by Jari27** [comment]
Ah that's fair. Not sure how I missed it! Thanks for the help!
